AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/AWSDataStorePluginTests.swift

```diff
@@ -61,6 +61,92 @@ class AWSDataStorePluginTests: XCTestCase {
         waitForExpectations(timeout: 1.0)
     }
 
+    func testStorageEngineStartsOnPluginStopStart() throws {
+        let stopExpectation = expectation(description: "Stop Sync should be called")
+        let startExpectation = expectation(description: "Start Sync should be called")
+        var currCount = 0
+
+        let storageEngine = MockStorageEngineBehavior()
+        storageEngine.responders[.stopSync] = StopSyncResponder { _ in
+            currCount = self.expect(stopExpectation, currCount, 1)
+        }
+
+        let storageEngineBehaviorFactory: StorageEngineBehaviorFactory = {_, _, _, _, _, _  throws in
+            return storageEngine
+        }
+        let dataStorePublisher = DataStorePublisher()
+        let plugin = AWSDataStorePlugin(modelRegistration: TestModelRegistration(),
+                                        storageEngineBehaviorFactory: storageEngineBehaviorFactory,
+                                        dataStorePublisher: dataStorePublisher,
+                                        validAPIPluginKey: "MockAPICategoryPlugin",
+                                        validAuthPluginKey: "MockAuthCategoryPlugin")
+        let finishNotReceived = expectation(
+            description: "publisher should not receive .finished completion event after stop() is called")
+        finishNotReceived.isInverted = true
+
+        do {
+            try plugin.configure(using: nil)
+            XCTAssertNil(plugin.storageEngine)
+
+            let sink = plugin.publisher.sink { completion in
+                switch completion {
+                case .finished:
+                    finishNotReceived.fulfill()
+                case .failure(let error):
+                    XCTFail("Error \(error)")
+                }
+            } receiveValue: { _ in }
+
+            storageEngine.responders[.startSync] = StartSyncResponder { _ in
+                currCount = self.expect(startExpectation, currCount, 2)
+            }
```

Comment:
does this need to be here, and not with the other storageEngine responder set up at line 70?

Reply:
not required, moved it up!